### PR TITLE
TASK: visual styling improvements

### PR DESCRIPTION
**Pull request by danielkestler**
Begin implementing UI improvements, see:
https://github.com/neos/guild-ux/issues/7

Just started cleaning some things up and working on the biggest flaws. Changes are:
- Make the tabs look like actual tabs now
- Add subtle border radius to all input elements for the inspector
- Correct the editor envelope highlight color (green does not make sense, changes are orange)
- Add a distinction between dropdown arrows and UI collapse arrows
- Change the ugly check icon to look nicer and use the icon component

More to follow.

Before:
ui-before

After:
ui-after

### CODE REVIEW

**Comment by dimaip** [comment]
Wow this is actually already looking real good, I can totally :+1: it.
We need to somehow make a process how to make such changes without turning it into "design by committee" situation. I'll ask around.

File: packages/neos-ui-editors/src/EditorEnvelope/style.css [hunk 1]
```diff
@@ -18,7 +18,9 @@
 }
 
 .envelope--highlight {
-    outline: 2px solid var(--colors-Success);
+    /* outline: 2px solid var(--colors-Success); */
```
Comment (dimaip) [inline packages/neos-ui-editors/src/EditorEnvelope/style.css]: Please don't leave commented code, just remove

Comment (danielkestler) [reply packages/neos-ui-editors/src/EditorEnvelope/style.css]: Oh, I am sorry.

File: packages/neos-ui-editors/src/EditorEnvelope/style.css [hunk 2]
```diff
@@ -18,7 +18,8 @@
 }
 
 .envelope--highlight {
-    outline: 2px solid var(--colors-Success);
+    box-shadow: 0 0 0 2px var(--colors-Warn);
+    border-radius: 2px;
```
Comment (bwaidelich) [inline packages/neos-ui-editors/src/EditorEnvelope/style.css]: Is the border radius chosen arbitrarily or was that used somewhere else already?

File: packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/NodeTreeSearchInput/style.css [hunk 3]
```diff
@@ -1,34 +1,30 @@
 .searchInput {
     flex-grow: 1;
-}
 
-.searchInput {
-    /* magic number, sorry */
-    padding-left: 32px;
+    input {
+        padding-left: 40px;
+        border-radius: 0;
```
Comment (bwaidelich) [inline packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/NodeTreeSearchInput/style.css]: you removed the "magic number, sorry" comment and replaced it with a new magic number :)
Can you elaborate?

File: packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/NodeTreeSearchInput/style.css [hunk 4]
```diff
@@ -1,34 +1,30 @@
...
-
 .wrapper {
     display: flex;
     border-right: 1px solid var(--colors-ContrastDark);
+    position: relative;
 }
 
 .hidden {
-    display: none;
+     display: none;
```
Comment (bwaidelich) [inline packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/NodeTreeSearchInput/style.css]: Is that new indention correct? Looks off to me

**Comment by danielkestler** [comment]
Someone did some weird things using some of the existing components. Will fix it.

File: packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/NodeTreeSearchInput/style.css (hunk 3 above)
Comment (dimaip) [reply packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/NodeTreeSearchInput/style.css]: There's already a constant for 40px I think.

But by any means, don't be side-tracked by code styling for now, we can nitpick you to death here :stuck_out_tongue: 
Besides other things, I'd propose to focus on the visual aspect first, it's far more important.

**Comment by danielkestler** [comment]
Because of the changes, the tests fail because of these errors:
expect(value).toMatchSnapshot()

Is there a trick to regenerate these snapshots?

**Comment by rolandschuetz** [comment]
From a visual point of few it's great! (Didn't check the code)

**Comment by danielkestler** [comment]
Started to give the publishing dropdown some love. Now represents the current publishing state.

bildschirmfoto 2019-02-01 um 08 09 54
bildschirmfoto 2019-02-01 um 08 10 08

**Comment by danielkestler** [comment]
Dropdowns in general now rock, BTW ;-)

bildschirmfoto 2019-02-01 um 08 14 24

**Comment by dimaip** [comment]
Really dope stuff!!!

While you are at it, scrollbar in dropdowns looks a bit off and untidy, wdyt?
image

Also the publish dropdown contents feels a bit divided from the button by the shadow. Maybe visually connect it with the dropdown toggle somehow?
image

**Comment by danielkestler** [comment]
Thank you. Yeah, all the scrollbars are not really nice ... can we influence the styling of them? They do not look like the browser ones. 

Will work on the publishing dropdown shadow.

**Comment by dimaip** [comment]
Yep you can style them in Chrome, here's how we do it: https://github.com/neos/neos-ui/blob/2239fe5465a84971d6aa6a94ea2f7a058959bf59/packages/neos-ui/src/Containers/style.css#L12

**Comment by danielkestler** [comment]
Neos started without a brand. Probably therefore, no logo has ever been added to the backend – I suggest to change this.

bildschirmfoto 2019-02-01 um 09 43 50
bildschirmfoto 2019-02-01 um 09 43 59

**Comment by jonnitto** [comment]
> Neos started without a brand. Probably therefore, no logo has ever been added to the backend – I suggest to change this.
> 
> bildschirmfoto 2019-02-01 um 09 43 50
> bildschirmfoto 2019-02-01 um 09 43 59

I'm not sure about this one. I always liked the "noiseless" backend of Neos. Perhaps just the icon?

**Comment by jonnitto** [comment]
> ... and while I'm working on it: numbers in brackets better be real badges, right?
> 
> bildschirmfoto 2019-02-01 um 08 22 20

What do you think about to align the badges to the right?

**Comment by danielkestler** [comment]
> What do you think about to align the badges to the right?

Will try that. The icons on the left are also not aligned correctly I think (it may be luck that they look aligned in the screenshot).

**Comment by danielkestler** [comment]
> I'm not sure about this one. I always liked the "noiseless" backend of Neos. Perhaps just the icon?
I'm open to discuss about it ;-) Personally, I like the logo there. If you only put the brand icon up there next to the menu icon, then there are two icons next to each other without any text, which looks a bit strange IMO – and I don't like to write "Menu" or something next to the icons.

**Comment by skurfuerst** [comment]
Hey @danielkestler ,

Just stumbled over this PR - and WOW - I *really* like your work :-) :-) :-) 

I am a bit unsure regarding the Logo (but that's probably because it's IMHO the most "drastic" of the changes and people are "Gewohnheitstiere". On the other hand, most other systems probably show you what you're working with ;)  

What can we do to get this change in? @danielkestler I would suggest instead of adding lots of new features to this PR, why not get this merged first and start new PRS? :) 

All the best, and keep up the awesome work,
Sebastian

**Comment by danielkestler** [comment]
... regarding last commit, here is a preview of the changes:

Before:
before

After:
after

**Comment by danielkestler** [comment]
> What can we do to get this change in? @danielkestler I would suggest instead of adding lots of new features to this PR, why not get this merged first and start new PRS? :)

Thank you, Sebastian. Of course I could "stop" here and wait for merging :) Well, I have a merge button here, but I am not sure if I am in the position to actually do this merge ... this is probably better kept at the reviewers.

**Comment by bwaidelich** [comment]
> Of course I could "stop" here and wait for merging

@danielkestler Please don't stop :)
But maybe you could cluster the changes a bit more into separate PRs – otherwise it's pretty impossible to review because you are so fast *g

But thanks again and keep up the good work!

**Comment by bwaidelich** [comment]
> maybe you could cluster the changes a bit more into separate PRs

not the existing ones of course, but changes to come

**Comment by dimaip** [comment]
> But maybe you could cluster the changes a bit more into separate PRs

I think it might make sense to extract "controversial" changes that might require more discussion (like the logo change), then let Daniel do as much of those smaller visual overhaul changes until he feels contend (or at least feels like it's a solid independent piece of work that can be merged) and when we are happy with the design side of those changes we can give it final technical review and merge.
Daniel, wdyt?

**Comment by danielkestler** [comment]
Of course, let's do it like that. So I guess I should remove the logo change again and put it into another PR?

**Comment by dimaip** [comment]
Yep, then it won't be blocking us for the final review

**Comment by danielkestler** [comment]
Okay here you go. Looking forward to you merging this :) ... if the tests pass successfully.

**Comment by danielkestler** [comment]
Added checkerboard for transparent images:

bildschirmfoto 2019-02-04 um 13 32 16

**Comment by aertmann** [comment]
Hey just came across this one again. This is turning into a monster pull request which should be avoided unless absolutely necessary. Instead push small individual PRs containing only minor contained adjustments. If you have to have dependent PRs then that's what you need to do. Otherwise it'll be difficult to review, discuss and reason about things and a bunch of small bugs will very likely slip through. Not to discourage you in any way, but consider the people you're asking to review your modifications as well.

**Comment by danielkestler** [comment]
Okay, will follow this next time. But IMO sometimes it does not make sense to separate things this way... regarding the UI changes: Would be strange to style e.g. only the dropdowns and make a PR, and then they look out of place. Everything is connected, right? Might have made more sense regarding the checkerboard pattern.

Sorry! Lucky for the reviewers, the changes are mostly CSS and now pass the tests ;-)

**Comment by dimaip** [comment]
Dear Daniel,
there's no problem at all with the way you've done things so far. While certainly it would be cool to extract bigger independent decisions into separate PRs, but for smaller related visual changes it's totally fine to have them in a single PR just as different commits.
For me it's actually easier to review such stuff as a single PR, as I don't need to jump back and forth between branches, constantly compiling stuff, which speeds things up considerably. If we'd find something problematic (as e.g. the logo change), we'd always be able to extract it into a different PR as needed.

So, bottom line:
- I'll do the final technical review on Friday and merge the PR then
- Till then feel free to push some more things into this PR
- If you plan bigger independent pieces of work, plan them as separate PRs (but based on 2.x, never make PRs dependent on each other)
- You are awesome!

**Comment by danielkestler** [comment]
Okay, that's it for now. Feel free to merge at any time.
The next changes maybe are too opinionated, so I will do them all in separate PRs after merging.

**Comment by jonnitto** [comment]
Wow, thanks again @danielkestler for all your work! 🏅 🏆 

I hope we'll have soon a release with all these nice improvements ❤️

**Comment by dimaip** [comment]
Merging this. Wanted to fix a few things, but don't have the merge rights. Will come up with a follow up.
